Sources/StreamChatUI/ChatChannelList/SwipeableView.swift

```diff
@@ -0,0 +1,178 @@
+
+    /// Constraint constant should be reset when view is being reused inside `UICollectionViewCell`.
+    public var actionStackViewWidthConstraint: NSLayoutConstraint?
+
+    /// `SwipeActionButtonsDelegate` instance
+    public weak var delegate: SwipeActionButtonsDelegate?
+
+    /// Value detecting start of the swipe gesture. We always start at 0
+    private var startValue: CGFloat = 0
+
+    /// The IndexPath is used in here to pass some reference for the given cell in action buttons closure.
+    /// we use this in delegate function calls `actionViews(forIndexPath)` and `willShowActionViews(forIndexPath)`
+    public var indexPath: IndexPath?
+
+    /// The `UIStackView` that arranges buttons revealed by swipe gesture.
+    open private(set) lazy var actionItemsStackView: UIStackView = uiConfig
+        .channelList
+        .swipeableViewSubviews
+        .actionButtonStack
+        .init()
+        .withoutAutoresizingMaskConstraints
+
+    override open func setUp() {
+        super.setUp()
+        panGestureRecognizer.delegate = self
+    }
+
+    override open func setUpLayout() {
+        super.setUpLayout()
+
+        addSubview(actionItemsStackView)
+        actionItemsStackView.pin(anchors: [.top, .trailing, .bottom, .height], to: self)
+
+        actionStackViewWidthConstraint = actionItemsStackView.widthAnchor.pin(equalToConstant: startValue)
+        actionStackViewWidthConstraint?.priority = .required
+        actionStackViewWidthConstraint?.isActive = true
+
+        actionItemsStackView.axis = .horizontal
+        actionItemsStackView.alignment = .fill
+        actionItemsStackView.distribution = .fillProportionally
+    }
+
+    @objc open func didPan(_ gesture: UIPanGestureRecognizer) {
+        // If we don't have indexPath or any actionViews, we don't want to proceed with the swiping.
+        guard let indexPath = indexPath, let actionButtons = delegate?.actionViews(for: indexPath) else { return }
+
+        if gesture.state == .began {
+            // If actionButtonStackView is not open, create the action buttons to show.
+            if !isStackViewOpen {
+                // Prepare the delegate that actionViews will be shown on some cell to
+                // reset other cells states or do some other cleanup.
+                delegate?.willShowActionViews(for: indexPath)
+
+                // Remove all subviews and add them to prevent having duplicities for views.
+                actionItemsStackView.removeAllArrangedSubviews()
+                actionButtons.forEach { actionItemsStackView.addArrangedSubview($0) }
+            }
+            startValue = actionStackViewWidthConstraint?.constant ?? 0
+        }
+
+        var xSwipePosition = gesture.translation(in: self).x
+
+        // If the language is Right to left, we need to switch the swipe direction, so we just negate it.
+        if !currentLanguageIsRightToLeftDirection {
+            xSwipePosition.negate()
```

Comment:
`value.negate()`: **exists**
Swift users: *lemme just put `-value` in here *

Actually I find this nicer (Strange argument comes) because this mutates instead of returning new value but I think it really doesn't matter :D